presto-testng-services/src/test/java/io/prestosql/testng/services/TestFlakyTestRetryAnalyzer.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.testng.services;
+
+import org.testng.annotations.Test;
+
+import static io.prestosql.testng.services.FlakyTestRetryAnalyzer.ALLOWED_RETRIES_COUNT;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+@Test(singleThreaded = true)
+public class TestFlakyTestRetryAnalyzer
```

Comment:
I suspect it's hard to write this kind of test but there's no check for the maximum retries count, ie. you do not check what's going to happen when the test fails 3 times.

Reply:
It is hard:

I tried with:
```
    @Flaky
    @Test(expectedExceptions = AssertionError.class, expectedExceptionsMessageRegExp = "3")
    public void testAlwaysFailingRetrying()
    {
        testAlwaysFailingRetryingCount++;
        fail("" + testAlwaysFailingRetryingCount);
    }
```

But such test just passes so it does not trigger the retrying so it is the same as `testRetrying`.